### DISCUSSION TITLE
Best way to separate mobile/web landing pages

### DISCUSSION THREAD

**Discussion by krishopper** [quasarframework/quasar#11182] [General - Components / Directives / etc] [answered]
I have a web app I developed in Quasar, and want to make a mobile version of the app.

I do not want to use a responsive version of the web app for the mobile version, as I want to customize the app home screen and some other things.

My question is: what is the best way to separate "Home" pages in a Quasar app for mobile versus web in the codebase?  These are two thoughts that came to mind, but I'm not sure if one of these is better than the other, or if there is a better way:

1. Modify routes.js to use a different set of routes if. That way I can easily have a different layout for mobile since I am using child routes; OR
2. Have the home page render a different child component based on the value of $q.capacitor

Thanks

**Comment by hawkeye64** [comment]
Personally, for this one, I'd use different routes. For the second one, it's likely you would have to load both pages, then use a variable to decide which one to display (via a `v-if` directive).

**Comment by suleimanas** [comment, marked as answer]
> 
> 
> I have a web app I developed in Quasar, and want to make a mobile version of the app.
> 
> I do not want to use a responsive version of the web app for the mobile version, as I want to customize the app home screen and some other things.
> 
> My question is: what is the best way to separate "Home" pages in a Quasar app for mobile versus web in the codebase? These are two thoughts that came to mind, but I'm not sure if one of these is better than the other, or if there is a better way:
> 
>     1. Modify routes.js to use a different set of routes if. That way I can easily have a different layout for mobile since I am using child routes; OR
> 
>     2. Have the home page render a different child component based on the value of $q.capacitor
> 
> 
> Thanks

1. Good, but need slightly more maintenance 
2. I do it a lot, same route and shared logic with different lazy-loaded components using v-if. like @hawkeye64 mentioned

**Comment by krishopper** [comment]
Thanks @hawkeye64 and @suleimanas. Sounds like my guess was along the right path then.
